service/src/main/kotlin/xyz/lebkuchenfm/api/xsounds/XSoundsEndpoint.kt

```diff
@@ -57,8 +59,9 @@ fun Route.xSoundsRouting(xSoundsService: XSoundsService) {
 
             fileBytes?.let { bytes ->
                 // TODO: pass authenticated user name as "addedBy"
-                val sound = xSoundsService.addNewXSound(soundName, tags, bytes)
-                call.respond(HttpStatusCode.Created, sound.toResponse())
+                xSoundsService.addNewXSound(soundName, tags, bytes)
+                    .onSuccess { call.respond(HttpStatusCode.Created, it.toResponse()) }
+                    .onFailure { call.respond(HttpStatusCode.InternalServerError, "Could not create new x-sound.") }
```

Comment:
We should have standardized error JSON response on REST endpoints. Maybe we should put this one in the doc and do a pass on all endpoints later?

Reply:
added to Doc